Home_Work13/HW13.cpp

```diff
@@ -0,0 +1,442 @@
+				std::string resultAfterGuess = intermidiateWordResult(inputString, guessWord, wordLength); //create intermidiate string
+
+				bool winCondition = checkWinCondition(guessWord, resultAfterGuess, wordLength); //check if the word is guessed
+
+				if (!winCondition) //if word no guessed than printing new round menu
+				{
+					printNewRoundMenu(inputString, resultAfterGuess, usedUnmatchLetters, currentAttemptsCounter);
+				}
+				else //if word guessed print win menu and update save file with true status if mode wa word of the day
+				{
+					printWinMenu(guessWord, maxAttempts, currentAttemptsCounter);
+					if (mainModeChoise == "2")
+					{
+						wordOfTheDayGuessStatus = "true";
+						writeSaveData(today, guessWord, wordOfTheDayGuessStatus, saveDataToFile);
+					}
+					break;
+				}
+			}
+		}
+
+	}
+}
+
+void printMainMenu()
+{
+	std::cout << std::endl;
+	std::cout << "!Hello this is Wordl Game!" << std::endl;
+	std::cout << "For start game in usual mode Enter 1" << std::endl;
+	std::cout << "For start game in Word of the day mode Enter 2" << std::endl;
+	std::cout << "For exit the game Enter 0" << std::endl;
+	std::cout << "Your input: ";
+}
+
+void printGameStartedMenu(int attempts)
+{
+	std::cout << std::endl;
+	std::cout << "Yor word generated try to guess it in " << attempts << " attempts" << std::endl;
+	std::cout << "To start game enter your guess" << std::endl;
+	std::cout << "To return to presious menu enter 0" << std::endl;
+	std::cout << "If the letter in result is uppercase this mean it correctlly guessed" << std::endl;
+	std::cout << "If the letter in result is lowercase this mean it in wrong place" << std::endl;
+}
+
+std::string mainModeErrors(std::string mainModeChoise)
+{
+	std::string inputErrors = "";
+	
+	bool inputContainsLettersOrSpecialChar = std::any_of(mainModeChoise.begin(), mainModeChoise.end(), [](char a) {if (!std::isdigit(a)) { return true; } return false; });
+	bool lengthNotEqualOne = (mainModeChoise.length() != 1);
+	bool wrongNumber = (mainModeChoise != "0" && mainModeChoise != "1" && mainModeChoise != "2");
+
+	if (lengthNotEqualOne)
+	{
+		inputErrors = " Input has wrong length ";
+	}
+
+	else if (inputContainsLettersOrSpecialChar)
+	{
+		inputErrors = " Input contain letters or special characters ";
+	}
+
+	else if (wrongNumber)
+	{
+		inputErrors = " Wrong control value ";
+	}
+
+	return inputErrors;
+}
+
+std::string gameModeErrors(std::string inputString, int wordLength)
+{
+	std::string inputErrors = "";
+	bool inputContainsDigitsOrSpecialChar = std::any_of(inputString.begin(), inputString.end(), [](char a) {if (!std::isalpha(a)) { return true; } return false; });
+	bool lengthNotEqualFive = (inputString.length() != wordLength);
+	bool lengthNotEqualOne = (inputString.length() != 1);
+	bool wrongNumber = (inputString != "0");
+
+	if (lengthNotEqualFive && lengthNotEqualOne)
+	{
+		inputErrors = " Input has wrong length ";
```

Comment:
tossing around strings like that just for error codes is not that free in terms of memory usages.
It's more common to create enum class ErrorCode {} with all codes you may encounter and then convert it into string const only when you really need to output this error to user